lib/app.js

```diff
@@ -108,6 +108,10 @@ const MAPPINGS = {
   genreId: ['ds:5', 1, 2, 79, 0, 0, 2],
   familyGenre: ['ds:5', 0, 12, 13, 1, 0],
   familyGenreId: ['ds:5', 0, 12, 13, 1, 2],
+  categories: {
```

Comment:
On first look, I'm not sure about this, it seems like you moved a lot of knowledge of the inner path to extract the field from here over to the helper function. Is that necessary? Ideally most of the path knowledge should reside in these declarative objects, and helper functions are mostly extraction logic.

If, for some reason, you need to put that knowledge in the helper it would be good to explain a bit what's going on in those function with some doc comments

Reply:
You're right, I do not like my implementation much, it's messy.. Submitted it only because it works for me somehow. 
Moved some logic from `mappingHelpers.js` to `app.js` MAPPINGS. 
Removed `familyGenre` and made that `genre` is added only when no `categories` are found, because that's what GP does it seems. 
Anyway, all these moves there are only to mimic closely Tags/Categories that are displayed visually on the GP page.